File: Problems/Algorithms/881. Boats to Save People/Boats.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        
        int ans = 0;

        int i = 0, j = people.length - 1;
        while (i < j) {
            if (people[i] + people[j] > limit) {
                j--;
                ans++;
            } else {
                j--;
                i++;
                ans++;
            }
        }
        
        if (i == j) ans++;
        
        return ans;
    }
}

File: Problems/Algorithms/881. Boats to Save People/Boats.kt
class Solution {
    fun numRescueBoats(people: IntArray, limit: Int): Int {
        people.sort()
        
        var ans = 0
        var i = 0
        var j = people.size-1
        while (i < j) {
            if (people[i] + people[j] > limit) {
                j--
                ans++
            } else {
                j--
                i++
                ans++
            }
        }
        
        if (i == j) ans++
        
        return ans
    }
}

